File: docs/guides/go/src/saveObjectsMovies.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func saveObjectsMovies() {
	// read json file
	url := "https://dashboard.algolia.com/api/1/sample_datasets?type=movie"

	response, err := http.Get(url)
	if err != nil {
		fmt.Println("Could not open url")

		return
	}

	defer response.Body.Close()

	// parse json file to Movie struct
	var movies []map[string]interface{}

	err = json.NewDecoder(response.Body).Decode(&movies)
	if err != nil {
		fmt.Println("Could not decode body")

		return
	}

	// initiate client with your app ID and write API key
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	// push data to algolia
	result, err := client.SaveObjects(context.Background(), "<YOUR_INDEX_NAME>", movies)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Done! Uploaded records in %d batches.", len(result))
}

File: docs/guides/csharp/src/saveObjectsMovies.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SaveObjectsMovies
{
  async Task Main(string[] args)
  {
    // read json file from url
    var url = "https://dashboard.algolia.com/api/1/sample_datasets?type=movie";
    var httpClient = new HttpClient();
    var response = await httpClient.GetAsync(url);
    var content = await response.Content.ReadAsStringAsync();

    // parse json
    var movies = JsonSerializer.Deserialize<List<dynamic>>(content);

    // initiate client and index with your app ID and write API key
    var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));

    // push data to algolia
    try
    {
      var result = await client.SaveObjectsAsync("<YOUR_INDEX_NAME>", movies);
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}
